### PR TITLE
[msys] use fresh msys, fix update core

### PR DESCRIPTION
**Pull request by isanych**
msys now using zstd packages not supported by old base (#11443), so using fresh base packages and fixing core update

### CODE REVIEW

**Comment by PhoebeHui** [comment]
Related to https://github.com/microsoft/vcpkg/pull/10782 and https://github.com/microsoft/vcpkg/pull/11455 and https://github.com/microsoft/vcpkg/pull/11443

**Comment by Voskrese** [comment]
updating MSYS itself to 20200517 solves this problem temporarily

the rest doesn't work, the update doesn't happen

```
error: cannot remove /usr/bin/pacman.exe (Permission denied)
warning: warning given when extracting /usr/bin/pacman.exe (Could not unlink)
```

**Comment by Voskrese** [comment]
wait for to removed symbolic links

https://github.com/microsoft/vcpkg/pull/11483

**Comment by isanych** [comment]
@Voskrese  That's why I run update twice, second times it finish successfully, and msys environment is usable in my testing, what exactly is not working?

**Comment by Voskrese** [comment]
problem with MSYS2 and software access symbolic links
it works locally, but not in VMSS

**Comment by isanych** [comment]
for me working with symlinkls too:
```
D:\w\v>mkdir downloads
D:\w\v>git clone -b msys2 https://github.com/isanych/vcpkg.git
D:\w\v>cd vcpkg
D:\w\v\vcpkg>mklink /D downloads ..\downloads
D:\w\v\vcpkg>bootstrap-vcpkg.bat
D:\w\v\vcpkg>vcpkg install icu
...
-- Acquiring MSYS2... OK
-- Acquiring MSYS Packages...
-- Acquiring MSYS Packages... OK
-- Configuring x86-windows-rel
-- Configuring x86-windows-rel done
-- Configuring x86-windows-dbg
-- Configuring x86-windows-dbg done
-- Package x86-windows-rel
-- Package x86-windows-rel done
-- Package x86-windows-dbg
-- Package x86-windows-dbg done
-- Warning: Could not find a matching pdb file for:
    D:/w/v/vcpkg/packages/icu_x86-windows/bin/icudt65.dll
    D:/w/v/vcpkg/packages/icu_x86-windows/debug/bin/icudtd65.dll

-- Installing: D:/w/v/vcpkg/packages/icu_x86-windows/share/icu/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package icu[core]:x86-windows... done
Installing package icu[core]:x86-windows...
Installing package icu[core]:x86-windows... done
Elapsed time for package icu:x86-windows: 5.017 min

Total elapsed time: 5.161 min

D:\w\v\vcpkg>dir
...
21/05/2020  09:23    <SYMLINKD>     downloads [..\downloads]
```

**Comment by Voskrese** [comment]
)))

**Comment by isanych** [comment]
the same log, just look further:
```
2020-05-20T20:31:13.1585124Z :: Processing package changes...
2020-05-20T20:31:13.1585905Z upgrading msys2-runtime...
2020-05-20T20:31:13.1586634Z error: cannot remove /usr/bin/msys-2.0.dll (Permission denied)
2020-05-20T20:31:13.1587171Z warning: warning given when extracting /usr/bin/msys-2.0.dll (Could not unlink)
2020-05-20T20:31:13.1588018Z warning: terminate MSYS2 without returning to shell and check for updates again
2020-05-20T20:31:13.1588562Z warning: for example close your terminal window instead of calling exit
2020-05-20T20:31:13.1589420Z :: Synchronizing package databases...
2020-05-20T20:31:13.1590163Z  mingw32 is up to date
2020-05-20T20:31:13.1590836Z  mingw64 is up to date
2020-05-20T20:31:13.1591668Z  msys is up to date
2020-05-20T20:31:13.1592291Z :: Starting core system upgrade...
2020-05-20T20:31:13.1593030Z  there is nothing to do
2020-05-20T20:31:13.1593809Z :: Starting full system upgrade...
2020-05-20T20:31:13.1594496Z resolving dependencies...
2020-05-20T20:31:13.1595402Z looking for conflicting packages...
2020-05-20T20:31:13.1596100Z 
2020-05-20T20:31:13.1597541Z Packages (10) bsdcpio-3.4.2-3  bsdtar-3.4.2-3  libarchive-3.4.2-3  libgnutls-3.6.13-2  libgpgme-1.13.1-4  libhogweed-3.6-1  libnettle-3.6-1  libpcre2_8-10.35-1  nettle-3.6-1  u-boot-tools-2020.04-2
2020-05-20T20:31:13.1598086Z 
2020-05-20T20:31:13.1598863Z Total Download Size:    5.67 MiB
2020-05-20T20:34:01.3964233Z Total Installed Size:  17.74 MiB
2020-05-20T20:34:01.3965417Z Net Upgrade Size:       0.03 MiB
2020-05-20T20:34:01.3965609Z 
2020-05-20T20:34:01.3965788Z :: Proceed with installation? [Y/n] 
2020-05-20T20:34:01.3966192Z :: Retrieving packages...
2020-05-20T20:34:01.3967280Z downloading libhogweed-3.6-1-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3967779Z downloading libnettle-3.6-1-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3968554Z downloading bsdcpio-3.4.2-3-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3969279Z downloading bsdtar-3.4.2-3-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3970671Z downloading libarchive-3.4.2-3-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3971156Z downloading libgnutls-3.6.13-2-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3972016Z downloading nettle-3.6-1-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3972762Z downloading libgpgme-1.13.1-4-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3973555Z downloading libpcre2_8-10.35-1-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3974231Z downloading u-boot-tools-2020.04-2-x86_64.pkg.tar.zst...
2020-05-20T20:34:01.3974987Z checking keyring...
2020-05-20T20:34:01.3975863Z checking package integrity...
2020-05-20T20:34:01.3976991Z loading package files...
2020-05-20T20:34:01.3977708Z checking for file conflicts...
2020-05-20T20:34:01.3978421Z checking available disk space...
2020-05-20T20:34:01.3979118Z :: Processing package changes...
2020-05-20T20:34:01.3979886Z upgrading libhogweed...
2020-05-20T20:34:01.3980415Z upgrading libnettle...
2020-05-20T20:34:01.3981308Z upgrading bsdcpio...
2020-05-20T20:34:01.3982180Z upgrading bsdtar...
2020-05-20T20:34:01.3983128Z upgrading libarchive...
2020-05-20T20:34:01.3983862Z upgrading libgnutls...
2020-05-20T20:34:01.3984844Z upgrading nettle...
2020-05-20T20:34:01.3985694Z upgrading libgpgme...
2020-05-20T20:34:01.3986453Z upgrading libpcre2_8...
2020-05-20T20:34:01.3987307Z upgrading u-boot-tools...
2020-05-20T20:34:01.3988101Z :: Synchronizing package databases...
2020-05-20T20:34:01.3989317Z  mingw32 is up to date
2020-05-20T20:34:01.3990054Z  mingw64 is up to date
2020-05-20T20:34:01.3990770Z  msys is up to date
2020-05-20T20:34:01.3991536Z :: Starting core system upgrade...
2020-05-20T20:34:01.3992253Z  there is nothing to do
2020-05-20T20:34:01.3993062Z :: Starting full system upgrade...
2020-05-20T20:34:01.3993827Z  there is nothing to do
2020-05-20T20:34:01.3994519Z -- Acquiring MSYS2... OK
```

**Comment by Voskrese** [comment]
```
2020-05-20T20:31:13.1586634Z error: cannot remove /usr/bin/msys-2.0.dll (Permission denied)
2020-05-20T20:31:13.1587171Z warning: warning given when extracting /usr/bin/msys-2.0.dll (Could not unlink)
2020-05-20T20:31:13.1588018Z warning: terminate MSYS2 without returning to shell and check for updates again
2020-05-20T20:31:13.1588562Z warning: for example close your terminal window instead of calling exit
2020-05-20T20:31:13.1589420Z :: Synchronizing package databases...
```

**core MSYS2 not updated**

**Comment by isanych** [comment]
It is updated during second run

**Comment by isanych** [comment]
```
2020-05-20T20:34:01.3990770Z  msys is up to date
```

**Comment by Voskrese** [comment]
`vcpkg remove icu && vcpkg install icu`

**Comment by Voskrese** [comment]
2020-05-20T20:34:01.4005802Z :: Proceed with installation? [Y/n] 

**2020-05-20T20:34:01** .4012035Z error: cannot remove /usr/bin/pacman.exe (Permission denied)
2020-05-20T20:34:01.4012783Z warning: warning given when extracting /usr/bin/pacman.exe (Could not unlink)


2020-05-20T20:34:01.4095330Z reinstalling pacman...
**2020-05-20T20:34:01** .4095880Z error: cannot remove /usr/bin/pacman.exe (Permission denied)
2020-05-20T20:34:01.4097544Z warning: warning given when extracting /usr/bin/pacman.exe (Could not unlink)

**Comment by isanych** [comment]
```
D:\w\v\vcpkg>vcpkg remove icu
The following packages will be removed:
    icu:x86-windows
Removing package icu:x86-windows...
Removing package icu:x86-windows... done

D:\w\v\vcpkg>vcpkg install icu
Your feedback is important to improve Vcpkg! Please take 3 minutes to complete our survey by running: vcpkg contact --survey
Computing installation plan...
The following packages will be built and installed:
    icu[core]:x86-windows
Starting package 1/1: icu:x86-windows
Building package icu[core]:x86-windows...
-- Using cached D:/w/v/vcpkg/downloads/icu4c-65_1-src.tgz
-- Using source at D:/w/v/vcpkg/buildtrees/icu/src/c-65_1-src-702c9cff96
-- Acquiring MSYS Packages...
-- Acquiring MSYS Packages... OK
-- Configuring x86-windows-rel
-- Configuring x86-windows-rel done
-- Configuring x86-windows-dbg
-- Configuring x86-windows-dbg done
-- Package x86-windows-rel
-- Package x86-windows-rel done
-- Package x86-windows-dbg
-- Package x86-windows-dbg done
-- Warning: Could not find a matching pdb file for:
    D:/w/v/vcpkg/packages/icu_x86-windows/bin/icudt65.dll
    D:/w/v/vcpkg/packages/icu_x86-windows/debug/bin/icudtd65.dll

-- Installing: D:/w/v/vcpkg/packages/icu_x86-windows/share/icu/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package icu[core]:x86-windows... done
Installing package icu[core]:x86-windows...
Installing package icu[core]:x86-windows... done
Elapsed time for package icu:x86-windows: 4.084 min

Total elapsed time: 4.084 min


D:\w\v\vcpkg>dir
21/05/2020  09:23    <SYMLINKD>     downloads [..\downloads]
```

**Comment by Voskrese** [comment]
I think it's a problem with access rights

**Comment by isanych** [comment]
Are you enabling native symlinks support in msys (`MSYS=winsymlinks:nativestrict`)? Still working fine for me on Win 10 2004 (10.0.19041.264) in developer mode - will tests in docker (I don't use VMs anymore) little bit later:
```
git config --global core.symlink true
set MSYS=winsymlinks:nativestrict
mkdir downloads
git clone -b msys2 https://github.com/isanych/vcpkg.git
cd vcpkg
mklink /D downloads ..\downloads
bootstrap-vcpkg.bat
vcpkg remove icu
vcpkg install icu
```

**Comment by Voskrese** [comment]
`sc qprivs gpsvc`  ?

**Comment by isanych** [comment]
That's old approach - I use developer mode in native windows and admin user in docker containers and the only issue I have with symlinks when they created in mapped directory of docker container running with hyperv isolation. Anyway I don't think any of symlink related business is related to this PR.

**Comment by isanych** [comment]
I had to kill gpg-agent, now it is working in docker too

**Comment by strega-nil** [comment]
This PR depends on #11483, so we'll merge it after that work is done.

**Comment by BillyONeal** [comment]
There's nothing about my symbolic links changes that would affect this -- in #11483 I'm fixing our ci command to not rely on symbolic links to get our temporary trees like 'installed' and 'buildtrees' onto a separate disk, I'm not changing anything about the ability to touch symbolic links from inside a port.

Notably, the CI machines run Windows Server, not Windows 10 Client, so the "allow dev without license" setting has no effect.

File: scripts/cmake/vcpkg_acquire_msys.cmake [hunk 1]
```diff
@@ -107,7 +119,7 @@ function(vcpkg_acquire_msys PATH_TO_ROOT_OUT)
     set(ENV{PATH} ${PATH_TO_ROOT}/usr/bin)
     vcpkg_execute_required_process(
       ALLOW_IN_DOWNLOAD_MODE
-      COMMAND ${PATH_TO_ROOT}/usr/bin/bash.exe --noprofile --norc -c "pacman -Sy --noconfirm --needed ${_am_PACKAGES}"
+      COMMAND ${PATH_TO_ROOT}/usr/bin/bash.exe --noprofile --norc -c "PATH=/usr/bin pacman -Sy --noconfirm --disable-download-timeout --needed ${_am_PACKAGES}"
```
Comment (BillyONeal) [inline scripts/cmake/vcpkg_acquire_msys.cmake:122]: I don't think --disable-download-timeout is OK. (Ditto elsewhere) If it's really taking so long to download these things we need to skip them in CI.

**Comment by BillyONeal** [comment]
It looks like the current msys breakage causes our builders to become wedged afterwards so we need to accept this or https://github.com/microsoft/vcpkg/pull/11443 after we've explicitly removed all the builders to get a clean pass.

**Comment by BillyONeal** [comment]
```
PS D:\vcpkg2> .\vcpkg install ffmpeg
Computing installation plan...
The following packages will be built and installed:
    ffmpeg[avresample,core]:x86-windows
  * zlib[core]:x86-windows
Additional packages (*) will be modified to complete this operation.
A suitable version of powershell-core was not found (required v6.2.1). Downloading portable powershell-core v6.2.1...
Downloading powershell-core...
  https://github.com/PowerShell/PowerShell/releases/download/v6.2.1/PowerShell-6.2.1-win-x86.zip -> D:\vcpkg2\downloads\PowerShell-6.2.1-win-x86.zip
Extracting powershell-core...
Starting package 1/2: zlib:x86-windows
Building package zlib[core]:x86-windows...
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
-- Extracting source D:/vcpkg2/downloads/zlib1211.tar.gz
-- Applying patch cmake_dont_build_more_than_needed.patch
-- Using source at D:/vcpkg2/buildtrees/zlib/src/1.2.11-deec42f53b
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: D:/vcpkg2/packages/zlib_x86-windows/share/zlib/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package zlib[core]:x86-windows... done
Installing package zlib[core]:x86-windows...
Installing package zlib[core]:x86-windows... done
Elapsed time for package zlib:x86-windows: 4.7 s
Starting package 2/2: ffmpeg:x86-windows
Building package ffmpeg[avresample,core]:x86-windows...
-- Downloading https://github.com/ffmpeg/ffmpeg/archive/n4.2.tar.gz...
-- Extracting source D:/vcpkg2/downloads/ffmpeg-ffmpeg-n4.2.tar.gz
-- Applying patch 0001-create-lib-libraries.patch
-- Applying patch 0003-fix-windowsinclude.patch
-- Applying patch 0004-fix-debug-build.patch
-- Applying patch 0005-fix-libvpx-linking.patch
-- Applying patch 0006-fix-StaticFeatures.patch
-- Using source at D:/vcpkg2/buildtrees/ffmpeg/src/n4.2-a4d149a5c6
-- Downloading https://www.tortall.net/projects/yasm/snapshots/v1.3.0.6.g1962/yasm-1.3.0.6.g1962.exe...
-- Installing: D:/vcpkg2/downloads/tools/yasm/1.3.0.6/yasm.exe
-- Acquiring MSYS2...
-- Downloading https://github.com/msys2/msys2-installer/releases/download/2020-05-17/msys2-base-x86_64-20200517.tar.xz...
gpg: /etc/pacman.d/gnupg/trustdb.gpg: trustdb created
gpg: no ultimately trusted keys found
gpg: starting migration from earlier GnuPG versions
gpg: porting secret keys from '/etc/pacman.d/gnupg/secring.gpg' to gpg-agent
gpg: migration succeeded
gpg: Generating pacman keyring master key...
gpg: key A613975912085877 marked as ultimately trusted
gpg: directory '/etc/pacman.d/gnupg/openpgp-revocs.d' created
gpg: revocation certificate stored as '/etc/pacman.d/gnupg/openpgp-revocs.d/BD3E563675C9E86CC3BFF4BAA613975912085877.rev'
gpg: Done
==> Updating trust database...
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
==> Appending keys from msys2.gpg...
==> Locally signing trusted keys in keyring...
  -> Locally signing key D55E7A6D7CE9BA1587C0ACACF40D263ECA25678A...
  -> Locally signing key 123D4D51A1793859C2BE916BBBE514E53E0D0813...
  -> Locally signing key B91BCF3303284BF90CC043CA9F418C233E652008...
  -> Locally signing key 9DD0D4217D75A33B896159E6DA7EF2ABAEEA755C...
==> Importing owner trust values...
gpg: setting ownertrust to 4
gpg: setting ownertrust to 4
gpg: setting ownertrust to 4
gpg: inserting ownertrust of 4
==> Updating trust database...
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   1  signed:   4  trust: 0-, 0q, 0n, 0m, 0f, 1u
gpg: depth: 1  valid:   4  signed:   3  trust: 0-, 0q, 0n, 4m, 0f, 0u
gpg: depth: 2  valid:   3  signed:   0  trust: 3-, 0q, 0n, 0m, 0f, 0u
SUCCESS: The process "gpg-agent.exe" with PID 16488 has been terminated.
:: Synchronizing package databases...
 mingw32 is up to date
 mingw64 is up to date
downloading msys.db...
downloading msys.db.sig...
:: Starting core system upgrade...
warning: resolving dependencies...
terminate other MSYS2 programs before proceeding
looking for conflicting packages...

Packages (2) msys2-runtime-3.1.4-2  pacman-5.2.1-7

Total Download Size:   16.80 MiB
Total Installed Size:  50.85 MiB
Net Upgrade Size:       1.05 MiB

:: Proceed with installation? [Y/n]
:: Retrieving packages...
downloading msys2-runtime-3.1.4-2-x86_64.pkg.tar.zst...
downloading pacman-5.2.1-7-x86_64.pkg.tar.zst...
ERROR: The process "gpg-agent.exe" not found.
:: Synchronizing package databases...
error: failed to update mingw32 (unable to lock database)
error: failed to update mingw64 (unable to lock database)
error: failed to update msys (unable to lock database)
error: failed to synchronize all databases
-- Acquiring MSYS2... OK
-- Acquiring MSYS Packages...
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: D:/vcpkg2/downloads/tools/msys2/msys64/usr/bin/bash.exe --noprofile --norc -c "PATH=/usr/bin pacman -Sy --noconfirm --disable-download-timeout --needed diffutils make"
    Working Directory: D:/vcpkg2/downloads/tools/msys2
    Error code: 1
    See logs for more information:
      D:\vcpkg2\buildtrees\ffmpeg\msys-pacman-x86-windows-out.log
      D:\vcpkg2\buildtrees\ffmpeg\msys-pacman-x86-windows-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_acquire_msys.cmake:123 (vcpkg_execute_required_process)
  ports/ffmpeg/portfile.cmake:41 (vcpkg_acquire_msys)
  scripts/ports.cmake:90 (include)


Error: Building package ffmpeg:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: ffmpeg:x86-windows
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above.
PS D:\vcpkg2>
```

`D:\vcpkg2\buildtrees\ffmpeg\msys-pacman-x86-windows-err.log` says:
```
error: failed to update mingw32 (unable to lock database)
error: failed to update mingw64 (unable to lock database)
error: failed to update msys (unable to lock database)
error: failed to synchronize all databases
```

Looks like CI isn't catching this right now...

**Comment by BillyONeal** [comment]
There are changes in here that are interesting in addition to the other PR we just merged, in particular updating our baseline version, so we still might take this one even though the other one made similar changes.

**Comment by isanych** [comment]
upstream removed 32 bit support, and using latest pacman in order better support future updates. Maybe it's time to let all 32 bit tools to go in vcpkg too. 32 bit triplets could be build on 64 bit host.

**Comment by strega-nil** [comment]
Given that #12467 was merged, closing :)

**Pull request closed without merging**